lib/utils/selection.js

```diff
@@ -50,4 +52,72 @@ exports.all = function (terminal) {
   }
 
   scrollPort.selection.sync();
-};
+}
+
+// select (highlight) between 2 indices of a row
+export function selectRow(terminal, row, startIndex, endIndex) {
+  const sel = terminal.document_.defaultView.getSelection();
+  const range = terminal.document_.createRange();
+  const rowNode = terminal.getRowNode(row);
+
+  // we need to consider that our indices might overflow the current row
+  const startNodeAndOffset = getNodeAndOffsetWithOverflow(rowNode, startIndex);
+  const endNodeAndOffset = getNodeAndOffsetWithOverflow(rowNode, endIndex);
+
+  if (startNodeAndOffset && endNodeAndOffset) {
+    range.setStart(startNodeAndOffset[0], startNodeAndOffset[1]);
+    range.setEnd(endNodeAndOffset[0], endNodeAndOffset[1]);
+    sel.removeAllRanges();
+    sel.addRange(range);
+  }
+}
+
+/*
+ * Copy-pasted from hterm's hterm.Screen.prototype.getNodeAndOffsetWithOverflow_
+ *
+ *
+ * Returns the node and offset corresponding to position within line.
+ * Supports line overflow.
+ *
+ * @param {Node} row X-ROW at beginning of line.
+ * @param {integer} position Position within line to retrieve node and offset.
+ * @return {Array} Two element array containing node and offset respectively.
+ */
+function getNodeAndOffsetWithOverflow(row, position) {
+  while (row && position > hterm.TextAttributes.nodeWidth(row)) {
+    if (row.hasAttribute('line-overflow') && row.nextSibling) {
+      position -= hterm.TextAttributes.nodeWidth(row);
+      row = row.nextSibling;
+    } else {
+      return -1;
+    }
+  }
+  return getNodeAndOffsetWithinRow(row, position);
+}
+
+/*
+ * Copy-pasted from hterm's hterm.Screen.prototype.getNodeAndOffsetWithinRow_
```

Comment:
I have to wonder if, rather than copying and pasting the code (duplicate code is a great way to make sure that a bug doesn't really get fixed), could you just use the function declared there?

something like
```
var GetNodeAndOffsetWithOverflow = hterm.Screen.prototype.getNodeAndOffsetWithOverflow_
```